File: ui/src/types/events.ts
// Config types
export interface Service {
  id: string;
  name: string;
  path: string;
  command: string;
  detect_url: boolean;
}

export interface Project {
  id: string;
  name: string;
  services: Service[];
}

export interface Config {
  version: number;
  active_project: string | null;
  projects: Project[];
}

// Log types
export interface LogEvent {
  source: string;
  level: 'normal' | 'error';
  text: string;
  timestamp: string;
  project_id: string;
}

// Status types
export interface ServiceStatus {
  running: boolean;
  url: string | null;
}

export interface StatusEvent {
  services: Record<string, ServiceStatus>;
}

// Filter types
export type LogLevel = 'normal' | 'error' | 'all';

export interface Filters {
  source: 'all' | 'system' | string; // string = service name
  level: LogLevel;
}

File: src-tauri/src/events.rs
use serde::Serialize;
use std::collections::HashMap;

#[derive(Clone, Serialize)]
pub struct LogEvent {
    pub source: String,
    pub level: String,
    pub text: String,
    pub timestamp: String,
    pub project_id: String,
}

#[derive(Clone, Serialize)]
pub struct ServiceStatus {
    pub running: bool,
    pub url: Option<String>,
}

#[derive(Clone, Serialize)]
pub struct StatusEvent {
    pub services: HashMap<String, ServiceStatus>,
}

#[derive(Clone)]
pub enum ManagerEvent {
    Log(LogEvent),
    Status(StatusEvent),
    ServiceUrl { service_id: String, url: String },
}
